docs/recipes/Code-Splitting-Using-Holocron.md

```diff
@@ -1,7 +1,238 @@
+    }}
+  />
+);
+```
+
+or can be an elaborate route configuration:
+
+```jsx
+import React from 'react';
+import { RenderModule } from 'holocron';
+import { ModuleRoute } from 'holocron-module-route';
+
+export const childRoutes = () => (
+  <ModuleRoute moduleName="holocron-reactor">
+    <ModuleRoute moduleName="holocron-reactor-hud" />
+    <ModuleRoute moduleName="holocron-reactor-beam" />
+    <ModuleRoute moduleName="holocron-reactor-engine">
+      <ModuleRoute moduleName="holocron-reactor-core" />
+    </ModuleRoute>
+  </ModuleRoute>
+);
+```
+
+## Holocron module chunks _via_ `import()`
+
+> To view a module chunks example using dynamic importing, [check out the sample module "Franks Burgers"][franks-burgers]
+
+With a Holocron module, we use code-splitting to add granularity to our modules and create chunks
+defined by user-driven actions or incorporate lazy loading for larger parts of our module.
+Let's start with a sample Holocron module project:
+
+```
+root
+|── locale
+|   └── en-US.json
+|── src
+|   ├── Module.jsx
+|   ├── Chunk.jsx
+|   └── index.js
+└── package.json
+```
+
+`src/Module.jsx`
+```jsx
+import React from 'react';
+
+export default function MyHolocronModule() {
+  const [Component, setComponent] = React.useState(null);
+
+  React.useEffect(() => {
+    import(/* webpackChunkName: "<chunkName>" */ './Chunk')
+      // the multi-line comment /* webpackChunkName: "..." */ is used by webpack to name your chunk
+      .then((importedChunk) => importedChunk.default || importedChunk)
+      // when we convert ES modules to common/supported formats,
+      // we might need to interop the `default` property to get the export
+      .then((chunk) => {
+        // ... do things with the chunk once we have what we want
+        setComponent(chunk);
+      });
+  }, []);
+
+  if (Component) return <Component />;
+
+  return null;
+}
+```
+
+`src/Chunk.jsx`
+```jsx
+import React from 'react';
+
+export default function ModuleChunk() {
+  return <p>My Holocron Module Chunk</p>;
+}
+```
+
+`src/index.js`
+```jsx
+import HolocronModule from './Module';
+
+export default HolocronModule;
+```
+
+In our basic example, if we wanted to use `Chunk.jsx` as a Holocron module chunk, we can use
+the supported `import()` syntax to divide up our module bundle into logical chunks and name
+our chunk using the _magic comments_ `/* webpackChunkName: "..." */`:
+
+```js
+// compared to traditional imports, which adds to the main bundle
+import './Chunk';
+
+import(/* webpackChunkName: "<chunkName>" */ './Chunk');
+```
+
+Our _chunk_ can be anything from a component to a `node_modules` package. Once we run `bundle-module` from `one-app-bundler`, our output in `build/<version>` will contain
+additional JavaScript files based on the number of chunks we have dynamically split
+from our module.
+
+```
+root
+└── build
+    └── 1.0.0
+        ├── <chunkName>.<moduleName>.chunk.browser.js
+        ├── vendors~<chunkName>.<moduleName>.chunk.browser.js
```

Comment:
I've never seen this before in a module. What causes it? Custom webpack config?

Reply:
Not a custom config, Webpack applies the default splitChunk configuration for dynamic importing, which splits up the vendors from the bundled script.

https://webpack.js.org/plugins/split-chunks-plugin/#optimizationsplitchunks